pkg/execution/state/redis_state/queue.go

```diff
@@ -985,46 +990,57 @@ func (q *queue) Peek(ctx context.Context, queueName string, until time.Time, lim
 	if err != nil {
 		return nil, err
 	}
-	items, err := scripts["queue/peek"].Exec(
+	res, err := scripts["queue/peek"].Exec(
 		ctx,
 		q.r,
 		[]string{
 			q.kg.QueueIndex(queueName),
 			q.kg.QueueItem(),
 		},
 		args,
-	).AsStrSlice()
+	).ToAny()
 	if err != nil {
 		return nil, fmt.Errorf("error peeking queue items: %w", err)
 	}
+	items, ok := res.([]any)
+	if !ok {
+		return nil, nil
+	}
+	if len(items) == 0 {
+		return nil, nil
+	}
 
-	// Create a slice up to items in length.  We're going to remove any items that are
-	// leased here, so we may end up returning less than the total length.
-	result := make([]*QueueItem, len(items))
-	n := 0
-	now := getNow()
-
-	for _, str := range items {
-		qi := &QueueItem{}
-		if err := json.Unmarshal([]byte(str), qi); err != nil {
-			return nil, fmt.Errorf("error unmarshalling peeked queue item: %w", err)
-		}
-		if qi.IsLeased(now) {
-			// Leased item, don't return.
-			continue
+	if isPeekNext {
+		i, err := q.decodeQueueItemFromPeek(items[0].(string), getNow())
+		if err != nil {
+			return nil, err
 		}
+		return []*QueueItem{i}, nil
+	}
 
-		// The nested osqueue.Item never has an ID set;  always re-set it
-		qi.Data.JobID = &qi.ID
-		result[n] = qi
-		n++
+	now := getNow()
+	return util.ParallelDecode(items, func(val any) (*QueueItem, error) {
+		return q.decodeQueueItemFromPeek(val.(string), now)
```

Comment:
probably should do a type check here?
seems like it'll panic in case it's not a string.

Reply:
done, ty :)